src/librustc_yk_sections/emit_tir.rs

```diff
@@ -337,8 +362,48 @@ impl<'tcx> ToPack<ykpack::Place> for (&ConvCx<'_, 'tcx, '_>, &Place<'tcx>) {
         let (ccx, place) = self;
 
         match place {
-            Place::Base(PlaceBase::Local(local)) => ykpack::Place::Local(ccx.tir_var(*local)),
-            _ => ykpack::Place::Unimplemented, // FIXME
+            Place::Base(pb) => ykpack::Place::Base((*ccx, pb).to_pack()),
+            Place::Projection(pj) => ykpack::Place::Projection((*ccx, pj.as_ref()).to_pack()),
+        }
+    }
+}
+
+/// Projection -> Pack
+impl<'tcx, T> ToPack<ykpack::Projection<Box<ykpack::Place>, ykpack::LocalIndex>>
+    for (&ConvCx<'_, 'tcx, '_>, &Projection<'tcx, Place<'tcx>, Local, T>)
+{
+    fn to_pack(&mut self) -> ykpack::Projection<Box<ykpack::Place>, ykpack::LocalIndex> {
+        let (ccx, pj) = self;
+
+        ykpack::Projection {
+            base: Box::new((*ccx, &pj.base).to_pack()),
+            elem: ykpack::ProjectionElem::Unimplemented(PhantomData), // FIXME
+        }
+    }
+}
+
+/// PlaceBase -> Pack
+impl<'tcx> ToPack<ykpack::PlaceBase> for (&ConvCx<'_, 'tcx, '_>, &PlaceBase<'tcx>) {
+    fn to_pack(&mut self) -> ykpack::PlaceBase {
+        let (ccx, pb) = self;
+
+        match pb {
+            PlaceBase::Local(local) => ykpack::PlaceBase::Local(ccx.tir_var(*local)),
+            PlaceBase::Static(s) => ykpack::PlaceBase::Static((*ccx, &s.as_ref().def_id).to_pack()),
+            PlaceBase::Promoted(bx) => ykpack::PlaceBase::Promoted(bx.0.as_u32()),
+        }
+    }
+}
+
+/// Operand -> Pack
+impl<'tcx> ToPack<ykpack::Operand> for (&ConvCx<'_, 'tcx, '_>, &Operand<'tcx>) {
+    fn to_pack(&mut self) -> ykpack::Operand {
+        let (ccx, op) = self;
+
+        match *op {
+            Operand::Move(place) | Operand::Copy(place)
+                => ykpack::Operand::Place((*ccx, place).to_pack()),
+            _ => ykpack::Operand::Unimplemented, // FIXME
```

Comment:
Bad formatting. Also I think we should make this `unimplemeted` shouldn't we? Or will that stop nearly every program from compiling?

Reply:
We don't want to crash, I just want to see an `unimplemented` node in my CFG.

Reply:
(I have a tool for drawing the cfg from a serialised pack)

Reply:
OK.

Reply:
WRT to formatting, how would you have done it? The second line is a continuation of the first, hence the indent. Joining the lines upsets the tidy check.

Reply:
Ah! Ignore me.